src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+                "nearest, nearest-exact, bilinear, bicubic, area."
+            )
+        if resample_method_output not in ("nearest", "nearest-exact", "bilinear", "bicubic", "area"):
+            raise ValueError(
+                "`resample_method_output` takes string values compatible with PIL library: "
+                "nearest, nearest-exact, bilinear, bicubic, area."
+            )
+        if batch_size < 0:
+            raise ValueError("`batch_size` must be non-negative: 0 for auto-detection or any positive value.")
+        if output_prediction_format not in ["pt", "np", "pil"]:
+            raise ValueError("`output_prediction_format` must be one of `pt`, `np`, or `pil`.")
+        if input_latent is not None and generator is not None:
+            raise ValueError("`input_latent` and `generator` are cannot be used together.")
+        if ensembling_kwargs is not None and not isinstance(ensembling_kwargs, dict):
+            raise ValueError("`ensembling_kwargs` must be a dictionary.")
+        if output_visualization_kwargs is not None and not isinstance(output_visualization_kwargs, dict):
+            raise ValueError("`output_visualization_kwargs` must be a dictionary.")
+
+        # memory saving hints
+        if save_memory:
+            logger.warning(
+                f"`save_memory` is currently not implemented. Consider setting `batch_size` to the minimum (1), "
+                f"ensure that `processing_resolution` is set to the recommended default value "
+                f"({self.optimal_processing_resolution}). Further memory savings can be achieved with pipeline "
+                f"offloading. Refer to https://huggingface.co/docs/diffusers/optimization/memory for more information."
+            )
+
+        def preset_override(new_denoising_steps: int, new_ensemble_size: int, new_processing_resolution: int):
+            nonlocal denoising_steps, ensemble_size, processing_resolution
+            if denoising_steps is not None:
+                logger.warning(
+                    f"Overriding `denoising_steps`={denoising_steps} due to preset {preset} with "
+                    f"value {new_denoising_steps}."
+                )
+            if ensemble_size is not None:
+                logger.warning(
+                    f"Overriding `ensemble_size`={ensemble_size} due to preset {preset} with value {new_ensemble_size}."
+                )
+            if processing_resolution is not None:
+                logger.warning(
+                    f"Overriding `processing_resolution`={processing_resolution} due to preset {preset} with "
+                    f"value {new_processing_resolution}."
+                )
+            denoising_steps = new_denoising_steps
+            ensemble_size = new_ensemble_size
+            processing_resolution = new_processing_resolution
+
+        def maybe_override(new_denoising_steps: int, new_ensemble_size: int, new_processing_resolution: int):
```

Comment:
Can we leverage the callback system do this? 
https://huggingface.co/docs/diffusers/main/en/using-diffusers/callback

This is how they are customized:
https://github.com/huggingface/diffusers/blob/0d7c4790235ac00b4524b492bc2a680dcc5cf6b0/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L161

Reply:
Sorry I don't understand how callbacks can help here. These functions are purely for convenience of automatically resolving presets, nothing is piped back to the caller here

Reply:
Fixed; all implicit logic is removed